ops/pebble.py

```diff
@@ -1969,6 +1983,95 @@ def read1(self, n: int = -1) -> Union[str, bytes]:
         return self.read(n)
 
 
+class IdentityAccess(enum.Enum):
+    """Enum of identity access levels."""
+
+    ADMIN = 'admin'
+    READ = 'read'
+    METRICS = 'metrics'
+    UNTRUSTED = 'untrusted'
+
+
+@dataclasses.dataclass
+class LocalIdentity:
+    """Local identity configuration (for ucrednet/UID authentication)."""
+
+    user_id: int
+
+    @classmethod
+    def from_dict(cls, d: LocalIdentityDict) -> LocalIdentity:
+        """Create new LocalIdentity from dict parsed from JSON."""
+        return cls(user_id=d['user-id'])
+
+    def to_dict(self) -> LocalIdentityDict:
+        """Convert this local identity to its dict representation."""
+        return {'user-id': self.user_id}
+
+
+@dataclasses.dataclass
+class BasicIdentity:
+    """Basic identity configuration (for HTTP basic authentication)."""
+
+    password: str
```

Comment:
Currently, the structure of the classes and the docs make it seem like you'll get a hash here when you do a query to Pebble to get identities, which would be bad. You actually get five `*`, which is good, but not explained anywhere here.

Is there any point having an attribute that will be `*****` for every identity? It's not a password, and it's not even a salted hash, it's just a fixed string. I don't see how this will be useful to anyone retrieving identities.

I feel this is worse when adding/updating identities get mixed in, because the same dict field is being used for a static mask string and also a salted hash - one has no security value at all and one is extremely high value, and they are being mixed into the same location.

Reply:
Hmm, yes, I think these are valid questions. However, this is the way the Pebble API works, so I think the method should and these types probably reflect that. However, we should document it better, something like "When writing, this is a SHA512-crypt hashed password. When reading, it is returned as `*****`."